src/Interfaces/AbstractZonotope.jl

```diff
@@ -525,3 +525,52 @@ function constraints_list(Z::AbstractZonotope{N}; check_full_rank::Bool=true
     end
     return constraints
 end
+
+"""
+    rectify(Z::AbstractZonotope{N}) where {N}
+
+Concrete rectification of a zonotopic set.
+
+### Input
+
+- `Z` -- zonotope
+
+### Output
+
+The zonotope obtained by rectifying `Z`.
+
+### Algorithm
+
+This function implements [Theroem 3.1, 1].
+
+[1] *Singh, G., Gehr, T., Mirman, M., Püschel, M., & Vechev, M. (2018). Fast
+and effective robustness certification. In Advances in Neural Information
+Processing Systems (pp. 10802-10813).*
+"""
+function rectify!(Z::AbstractZonotope{N}) where {N}
```

Comment:
This function may be misleading because it modifies the zonotope but it still returns a new one. I would have expected that no new zonotope is allocated.

Reply:
good point.  we should either modify and return `Z` or change the signature to `rectify!(Zout, Z)`. i prefer the former if possible